src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE = 1.5;
+    private Object[] elementsData;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elementsData = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkGrow();
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkGrow();
+        checkIndexAdd(index);
+        System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+        elementsData[index] = value;
+        size++;
+    }
+
+    public void checkIndexGet(int index) {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index: " + index);
+        }
+    }
 
+    public void checkIndexAdd(int index) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index: " + index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndexGet(index);
+        return (T) elementsData[index];
     }
 
     @Override
     public void set(T value, int index) {
+        checkIndexGet(index);
+        elementsData[index] = value;
+    }
 
+    public void checkGrow() {
+        if (elementsData.length == size) {
+            grow();
+        }
+    }
+
+    public void grow() {
+        int currentCapacity = elementsData.length;
+        int newCapacity = (int) Math.round(currentCapacity * RESIZE);
+        elementsData = Arrays.copyOf(elementsData, newCapacity);
     }
 
```

Comment:
use more informative name for this method as described [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#use-informative-names-for-your-variables-and-methods)